fftvis/utils.py

```diff
@@ -65,3 +65,51 @@ def get_pos_reds(antpos, decimals=3, include_autos=True):
             reds_list.append(red)
 
     return reds_list
+
+
+def get_plane_to_xy_rotation_matrix(antvecs):
+    """
+    Compute the rotation matrix that projects the antenna positions onto the xy-plane. 
+    This function is used to rotate the antenna positions so that they lie in the xy-plane.
+
+    Parameters:
+    ----------
+        antvecs: np.array
+            Array of antenna positions in the form (Nants, 3).
+
+    Returns:
+    -------
+        rotation_matrix: np.array
+            Rotation matrix that projects the antenna positions onto the xy-plane of shape (3, 3).
+    """
+    # Fit a plane to the antenna positions
+    antx, anty, antz = antvecs.T
+    basis = np.array([antx, anty, np.ones_like(antz)]).T
+    plane, res, rank, s = linalg.lstsq(basis, antz)
+    
+    # Project the antenna positions onto the plane
+    slope_x, slope_y, z_offset = plane
+
+    # Plane is already approximately aligned with the xy-axes, 
+    # return identity rotation matrix
+    if np.isclose(slope_x, 0) and np.isclose(slope_y, 0.0):
+        return np.eye(3)
+
+    # Normalize the normal vector
```

Comment:
This looks like it probably works, but I also had gotten a working solution that might be a little more compact (maybe faster, not sure):

```python
from scipy.spatial.transform import Rotation
mean = np.mean(antvecs, axis=0)
svd = np.linalg.svd(antvecs.T - mean[None].T)

# Get the normal vector of the plane
normal = svd[0][:, -1]

cross = np.cross(np.array([0,0,1]), normal)
rot = Rotation.from_rotvec(cross)
return rot.as_matrix()
```

Reply:
Or you could just return `rot` itself, and use `rot.apply(antvecs)` to get new antenna vectors that are rotated to the plane (they will still have whatever offset mean they started with).

Reply:
I like the compactness of the solution, but I'm finding that it's not reducing the z-component as the existing solution. Maybe I'm doing something wrong here?

<img width="928" alt="Screenshot 2024-08-23 at 12 18 20 PM" src="https://github.com/user-attachments/assets/d01d2f53-6b4c-440a-8335-170f222c8333">

Reply:
I would think the mean abs of the z-component would actually be expected to be about 12, after rotation, because at least in the compact method, it only applies a rotation around the mean point, which means the output points should all have z-components equivalent to that mean. Here it looks like that mean should be in the ballpark of 12. You can also apply an offset to bring the points back down to the z=0 plane, but I don't think that is necessary.